content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,825 @@
+compute Chi-square tests for each table, using `lapply`; note that we show
+only the first few tests:
+
+```{r stegen_food_tests, echo = c(1,3), eval = -2}
+food_tests <- lapply(food_tables, chisq.test)
+food_tests
+head(food_tests)
+```
+
+Looking at these results, it seems a large number of food items are
+significantly correlated to the illness. Which food item is the biggest suspect?
+We will try and address this question using risk ratios.
+
+
+
+## Risk ratios
+
+### For one variable 
+
+The [risk ratio](https://en.wikipedia.org/wiki/Risk_ratio) is defined as the
+ratio between the proportion of illness in one group (typically 'exposed') vs
+another ('non-exposed'). For instance, let us consider the contingency table of
+pork consumption and illness:
+
+```{r stegen_pork_table}
+food_tables$pork
+```
+
+This can be turned into proportions using `prop.table`, specifying that we want
+proportions by rows (`margin = 1`):
+
+```{r stegen_pork_prop}
+pork_prop <- prop.table(food_tables$pork, margin = 1)
+pork_prop
+```
+
+the corresponding risk ratio is:
+
+```{r stegen_pork_rr}
+## ill & exposed: row 2, column 2
+pork_prop[2, 2] # row 2, column 2
+
+## ill & non-exposed: row 1, column 2
+pork_prop[1, 2]
+
+## compute the risk ratio
+pork_rr <- pork_prop[2, 2] / pork_prop[1, 2]
+pork_rr
+```
+
+Note the use of the syntax `[rows, columns]` to subset specific rows and columns
+of 2-dimensional objects.
+
+
+### Multiple variables
+
+In the example above, we have computed the risk ratio using a simple recipe:
+
+1. select a contingency table
+2. convert raw numbers to proportions (using `prop.table`)
+3. isolating the risks in exposed and non-exposed groups, and compute their
+   ratio
+
+Repeating these tasks manually for all variables in `food` would be
+cumbersome. As an alternative, we can define the steps above as `function`,
+i.e. a generic recipe which we can then apply to any contingency table. This new
+function will be called `risk_ratio`, and is defined as follows:
+
+```{r stegen_risk_ratio_function}
+risk_ratio <- function(tab) { # 'tab' is a placeholder for the input 2x2 table
+  ## compute table of proportions
+  tab_prop <- prop.table(tab, margin = 1)
+  
+  ## compute the risk ratio
+  result <- tab_prop[2, 2] / tab_prop[1, 2]
+
+  ## return the output
+  return(result) # this will be the output of the function
+}
+```
+
+We can now try this function on any of the tables in `food_tables`, for instance:
+
+```{r rr_demo}
+food_tables$pork
+risk_ratio(food_tables$pork)
+```
+
+And we can now apply this function sequentially to all variables in `food` as
+before, using `lapply` (we show only the first few values):
+
+```{r stegen_all_rr, echo = 1:2, eval = -2}
+all_rr <- lapply(food_tables, risk_ratio)
+all_rr
+head(all_rr)
+```
+
+Finally, we re-shape these results into a numeric `vector` (using `unlist)` for
+further plotting; note that for this graph, the basic system is easier to use
```

Comment:
I think we should stick with ggplot2. If we go the barplot route, we have to also explain base plotting including par(), mar, las, and lty.

The only difficulty is that ggplot2 isn't great for sideways barplots. We could (a) have a vertical barplot or (b) have a dot and line plot. I've coded up the latter:

```r
rr_data <- unlist(all_rr)
rr_data <- sort(rr_data, decreasing = TRUE)
rr_df    <- stack(rr_data)
ggplot(rr_df, aes(x = values, y = ind)) + 
  geom_point() + 
  geom_vline(xintercept = 1, lty = 2) + 
  geom_segment(aes(xend = 1, yend = ind)) +
  theme_light(base_size = 16, base_family = "Times") +
  labs(x = "Risk Ratio", 
       y = "Food Item", 
       title = "Risk Ratio by Food Exposure")
```

![image](https://user-images.githubusercontent.com/3639446/47574058-aedd4c00-d936-11e8-8ab5-a4b2cf4ea800.png)

Reply:
OK I'm fine with this, and the graph looks good. Maybe just explain quickly what `stack` is?